### ISSUE TITLE
Train Model

### ISSUE THREAD

**Issue by shwe87** [AMLResearchProject/ALL-Pytorch-2021#11]
The challenge will be to get the correct hyperparameters to get the best model as in learning rate, optimizer, number of layers, etc. These parameters should be part of the training method.

While training, validation should also be done so that the training should be stopped before it starts overfitting.

** Please take in mind that the timelines are long for this phase because phases 2, 3, and 4 should be done in parallel to get the optimized model.
